spec/controllers/idv/doc_auth_controller_spec.rb

```diff
@@ -303,15 +314,44 @@
       expect(response.body).to eq(
         {
           success: false,
-          errors: [{ field: 'pii',
-                     message: I18n.t('doc_auth.errors.general.no_liveness') }],
+          errors: [{ field: 'front', message: 'Wrong document' }],
           remaining_attempts: IdentityConfig.store.doc_auth_max_attempts,
           ocr_pii: nil,
+          result_failed: true,
         }.to_json,
       )
-      expect(@analytics).to have_received(:track_event).with(
-        'IdV: ' + "#{Analytics::DOC_AUTH} verify_document_status submitted".downcase, {
-          errors: { pii: [I18n.t('doc_auth.errors.general.no_liveness')] },
+    end
+
+    it 'returns status of fail with incomplete PII from doc auth' do
+      set_up_document_capture_result(
+        uuid: document_capture_session_uuid,
+        idv_result: bad_pii_result,
+      )
+
+      expect(@analytics).to receive(:track_event).with(
+        'IdV: ' + "#{Analytics::DOC_AUTH} image upload vendor pii validation".downcase, include(
+          errors: include(
+            pii: satisfy do |v|
+              expect(v&.[](0)).to eq(I18n.t('doc_auth.errors.general.no_liveness'))
+            end,
+          ),
+          error_details: { pii: [I18n.t('doc_auth.errors.general.no_liveness')] },
+          attention_with_barcode: false,
+          success: false,
+          remaining_attempts: IdentityConfig.store.doc_auth_max_attempts,
+          flow_path: 'standard',
+          pii_like_keypaths: [[:pii]],
+          user_id: nil,
+        )
+      )
+
+      expect(@analytics).to receive(:track_event).with(
+        'IdV: ' + "#{Analytics::DOC_AUTH} verify_document_status submitted".downcase, include(
+          errors: include(
+            pii: satisfy do |v|
+              expect(v&.[](0)).to eq(I18n.t('doc_auth.errors.general.no_liveness'))
```

Comment:
would this work too?
```suggestion
              expect(v&.first).to eq(I18n.t('doc_auth.errors.general.no_liveness'))
```
(and the same in the other places where `v&.[](0)` is used?

Reply:
These revisions were added by @NavaTim , but I'd guess if we're expecting a specific error, we could remove the nullish operator altogether? I'll give it a shot.

Reply:
Updated in 17f6e1d1f.